modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/DataStreamIT.java

```diff
@@ -2441,7 +2441,13 @@ static void putComposableIndexTemplate(
         request.indexTemplate(
             ComposableIndexTemplate.builder()
                 .indexPatterns(patterns)
-                .template(new Template(settings, mappings == null ? null : CompressedXContent.fromJSON(mappings), aliases, lifecycle))
+                .template(
+                    Template.builder()
+                        .settings(settings)
+                        .mappings(mappings == null ? null : CompressedXContent.fromJSON(mappings))
+                        .aliases(aliases)
+                        .lifecycle(lifecycle)
```

Comment:
Nit: Is the builder helping here?

Reply:
Good point, not necessarily because we are setting all the values here. I can revert this